### ISSUE TITLE
Stack overflow with overlapping polygons

### ISSUE THREAD

**Issue by Siccity** [karl-/pb_CSG#6]
I'm subtracting a simple box from a larger, simple box. I think the problem is, the smaller box is completely inside the larger box, but one side is overlapping (think z-fighting) with the outer box. Not sure if this is the issue, but it could be.

```
StackOverflowException: The requested operation caused a stack overflow.
Parabox.CSG.CSG_Node.Build (System.Collections.Generic.List`1 list) (at Assets/T3DImporter/Submodules/pb_CSG-master/CSG/Classes/CSG_Node.cs:126)
Parabox.CSG.CSG_Node.Build (System.Collections.Generic.List`1 list) (at Assets/T3DImporter/Submodules/pb_CSG-master/CSG/Classes/CSG_Node.cs:126)
Parabox.CSG.CSG_Node.Build (System.Collections.Generic.List`1 list) (at Assets/T3DImporter/Submodules/pb_CSG-master/CSG/Classes/CSG_Node.cs:126)
Parabox.CSG.CSG_Node.Build (System.Collections.Generic.List`1 list) (at Assets/T3DImporter/Submodules/pb_CSG-master/CSG/Classes/CSG_Node.cs:126)
Parabox.CSG.CSG_Node.Build (System.Collections.Generic.List`1 list) (at Assets/T3DImporter/Submodules/pb_CSG-master/CSG/Classes/CSG_Node.cs:126)
```

**Comment by DanSh21** [comment]
Same issue when rotation of object not (0,0,0)

**Comment by tuseroni** [comment]
i had this same issue, what i did to fix it was at line 112 of csg_node.cs i changed:

    if (list_front.Count > 0)

to 

    if (list_front.Count > 0 && list_front.Count != list.Count)

and for line 120 i changed 

    if (list_back.Count > 0)

to

    if (list_back.Count > 0 && list_back.Count != list.Count)

after this the subtraction should work but your scale and centre might be off, there is another issue here where i posted code to handle that.

i'd make a pull request but the github desktop app seems to be broken, perhaps when i get into linux i can make a PR.

**Comment by Pequ88** [comment]
In my case the EPSILON value was too low. I gues a floating point inaccuracy was enough to trigger every Coplanar polygons to be front polygons.
`EPolygonType type = (t < -CSG.EPSILON) ? EPolygonType.Back : ((t > CSG.EPSILON) ? EPolygonType.Front : EPolygonType.Coplanar);`
So I just raised the `CSG.EPSILON` value to .0001f from 0.00001f
Bear in mind, all my meshes are cubes, so this may break the functionality on some more complex meshes.

**Comment by karl-** [comment]
Should be fixed as of e2d2e1d, please re-open if not addressed.
